src/app/providers/file-drop-provider.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {
+  createContext,
+  useState,
+  type ReactNode,
+  type RefObject,
+} from "react";
+
+interface FileDropProviderProps {
+  children: ReactNode;
+  inputRef: RefObject<HTMLInputElement | null>;
+}
+
+interface FileDropContextType {
+  isDragging: boolean;
+}
+
+const FileDropContext = createContext<FileDropContextType>({
+  isDragging: false,
+});
+
+export const FileDropProvider: React.FC<FileDropProviderProps> = ({
+  children,
+  inputRef,
+}) => {
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleDragOver = (e: React.DragEvent) => {
+    e.preventDefault();
+    setIsDragging(true);
+  };
+
+  const handleDragLeave = (e: React.DragEvent) => {
+    e.preventDefault();
+    setIsDragging(false);
+  };
+
+  const handleDrop = (e: React.DragEvent) => {
+    e.preventDefault();
+    setIsDragging(false);
+
+    if (e.dataTransfer.files && e.dataTransfer.files.length > 0) {
+      const file = e.dataTransfer.files[0];
+
+      if (!file) return;
+
+      // Check if the file type matches the allowed types in the input's accept attribute
+      const accept = inputRef.current?.accept;
+      const allowedTypes = accept ? accept.split(",") : [];
+      const isFileTypeAllowed = allowedTypes.some((type) => {
+        type = type.trim();
+        return (
+          file.type === type ||
+          (type === "image/*" && file.type.startsWith("image/")) ||
+          (type.startsWith(".") && file.name.endsWith(type))
+        );
+      });
+
+      if (isFileTypeAllowed && inputRef.current) {
+        inputRef.current.files = e.dataTransfer.files;
+
+        // Trigger the change event on the input to update the file input's state
+        const event = new Event("change", { bubbles: true });
+        inputRef.current.dispatchEvent(event);
+        e.dataTransfer.clearData();
+      } else {
+        alert("File type not allowed. Please upload a valid file.");
```

Comment:
Instead of a simple alert, make a popup which is better UI

Reply:
There is no alert system implemented in the app at the moment so adding this will be out of this PR's scope. That can be a feature PR in itself